### ISSUE TITLE
Bun throws error when copying file to a folder that does not exist

### ISSUE THREAD

**Issue by birkskyum** [oven-sh/bun#11260]
### What version of Bun is running?

1.1.10-canary.1+cc0bce62e

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

I experience that I can make a clean `basic` example (bun create solid), dump a few files in the public folder, like .glb 3d models, and then the `bun --bun run build` command will break with:

```sh
 ERROR  { [ENOENT: No such file or directory]                                                                                                            9:06:21 AM
  code: 'ENOENT',
  path:
   '/Users/admin/repos/bun-kitchensink/solid/wefwef/public/3dmodels/giantherb/giantherb.glb',
  syscall: 'copyfile',
  errno: -2 } 



 ERROR  No such file or directory                                                                                                                        9:06:21 AM

error: script "build" exited with code 1
```

It's worth noting that it's not a requirement to use these files anywhere in the code for this to break

### What is the expected behavior?

When running `pnpm run build` or npm, then the build completes

### Additional information

Related to
- #3736

**Comment by posterhoudt** [comment]
Confirmed, @birkskyum. I have the same issue (used Bun v1.1.9 on Windows 11):

![image](https://github.com/oven-sh/bun/assets/163450497/71b1778e-7db6-4230-8412-8ea34cba0845)

**Comment by zoto-ff** [comment]
the error occurs when working with nitropack (vinxi/nitro component). something is wrong with fs.promises.cp(). its occurs when performing all copies in parallel (await Promise.all(...)) and in queue (for (...) { await ... })

![IMG_6463](https://github.com/oven-sh/bun/assets/142039751/94348ac0-e239-4d03-9dca-b84b6e24805a)

https://github.com/unjs/nitro/blob/61f2079a605fca81b578894edfdbb0c328db12a3/src/build.ts#L87

**Comment by sirenkovladd** [comment]
Smaller reproduce

```js
import { promises } from 'node:fs';

const src = 'newFile.txt';
const folder = 'folder-not-exist'
const dst = `${folder}/${src}`;

await promises.writeFile(src, 'A'.repeat(131073))

await promises.rm(folder, { recursive: true, force: true })
await promises.cp(src, dst);
console.log('Success')
```

```sh
❯ bun test.js 
ENOENT: No such file or directory
   errno: -2
 syscall: "copyfile"
   path: "newFile.txt"


Bun v1.1.10 (macOS arm64)

❯ node test.js
Success
```

**Comment by birkskyum** [comment]
Now that it's much more narrowed down - What would be a better title for this issue?

**Comment by posterhoudt** [comment]
I didn't use .repeat when I got the error (unless SolidStart is using it internally, which I doubt.).

**Comment by Huliiiiii** [comment]
If folder doesn't exist, cp in node will create folder, but bun won't.

**Comment by sirenkovladd** [comment]
Bun error occurs if you need to copy a `large` file to a folder that does not exist
The problem is precisely only in the large file, since everything is fine with small files

**Comment by posterhoudt** [comment]
So @birkskyum, new issue title could be, "Bun throws error when copying large file to a folder that does not exist"?

**Comment by posterhoudt** [comment]
Thanks to @sirenkovladd for supplying the "smaller reproduce"!

**Comment by Huliiiiii** [comment]
> 如果您需要将`large`文件复制到不存在的文件夹，则会出现 Bun 错误 问题恰恰只出现在大文件中，因为小文件一切正常

```
await promises.writeFile(src, 'A')
```
On my computer, this also causes the error, so I don't think the file size is the issue.

**Comment by Huliiiiii** [comment]
https://github.com/oven-sh/bun/blob/99b979e5caaa0052e79eb4b5d8eb3ea11702dbcf/src/js/node/fs.promises.ts#L4
https://github.com/oven-sh/bun/blob/99b979e5caaa0052e79eb4b5d8eb3ea11702dbcf/src/js/node/fs.promises.ts#L97-L109

Set any option that makes this function return the js fallback ported from node will make the program succeed.
So I think the problem is in zig's implementation.

**Comment by birkskyum** [comment]
Nice find with the repro @Huliiiiii . Seems like some zig expertise is due at this point. cc @nektro
